hydra-node/src/Hydra/Chain/Blockfrost/Client.hs

```diff
@@ -277,22 +227,167 @@ toCardanoPParams = do
+              Just datumHash -> TxOutDatumHash (fromString $ T.unpack datumHash)
+          Just cborDatum ->
+            case deserialiseFromCBOR (proxyToAsType (Proxy @HashableScriptData)) (encodeUtf8 cborDatum) of
+              Left _ -> TxOutDatumNone
+              Right hashableScriptData -> TxOutDatumInline hashableScriptData
+  case plutusScript of
+    Nothing -> do
+      case toCardanoAddress addrTxt of
+        Nothing -> liftIO $ throwIO $ BlockfrostError $ "Failed to decode address: " <> addrTxt
+        Just addr -> pure $ TxOut addr val datum ReferenceScriptNone
+    Just script -> pure $ TxOut (scriptAddr script) val datum (mkScriptRef script)
+ where
+  scriptAddr script =
+    makeShelleyAddressInEra
+      shelleyBasedEra
+      networkId
+      (PaymentCredentialByScript $ hashScript $ PlutusScript script)
+      NoStakeAddress
+
+toCardanoPolicyIdAndAssetName :: Text -> BlockfrostClientT IO (PolicyId, AssetName)
+toCardanoPolicyIdAndAssetName pid = do
+  Blockfrost.AssetDetails{_assetDetailsPolicyId, _assetDetailsAssetName} <- Blockfrost.getAssetDetails (Blockfrost.mkAssetId pid)
+  case deserialiseFromRawBytesHex (encodeUtf8 $ Blockfrost.unPolicyId _assetDetailsPolicyId) of
+    Left err -> liftIO $ throwIO $ BlockfrostError $ show err
+    Right p ->
+      case _assetDetailsAssetName of
+        Nothing -> liftIO $ throwIO $ BlockfrostError "Asset name is missing."
+        Just assetName ->
+          case deserialiseFromRawBytesHex (encodeUtf8 assetName) of
+            Left err -> liftIO $ throwIO $ BlockfrostError $ show err
+            Right asset -> pure (p, asset)
+
+toCardanoValue :: [Blockfrost.Amount] -> BlockfrostClientT IO Value
+toCardanoValue = foldMapM convertAmount
+ where
+  convertAmount (Blockfrost.AdaAmount lovelaces) =
+    pure $
+      fromList
+        [
+          ( AdaAssetId
+          , Quantity (toInteger lovelaces)
+          )
+        ]
+  convertAmount (Blockfrost.AssetAmount money) = do
+    let currency = Money.someDiscreteCurrency money
+    (cardanoPolicyId, assetName) <- toCardanoPolicyIdAndAssetName currency
+    pure $
+      fromList
+        [
+          ( AssetId
+              cardanoPolicyId
+              assetName
+          , Quantity (Money.someDiscreteAmount money)
+          )
+        ]
+
+addressToText :: AddressInEra -> Text
+addressToText = \case
+  ShelleyAddressInEra addr -> serialiseToBech32 addr
+  ByronAddressInEra{} -> error "Byron."
+
+toCardanoAddress :: Text -> Maybe AddressInEra
+toCardanoAddress addrTxt =
+  ShelleyAddressInEra <$> deserialiseAddress (AsAddress AsShelleyAddr) addrTxt
+
+textAddrOf :: NetworkId -> VerificationKey PaymentKey -> Text
+textAddrOf networkId vk = addressToText (mkVkAddress @Era networkId vk)
+
+toTxHash :: TxId -> Text
+toTxHash (TxId hash) = hashToTextAsHex hash
+
+toCardanoNetworkId :: Integer -> NetworkId
+toCardanoNetworkId = \case
+  0 -> Mainnet
+  magicNbr -> Testnet (NetworkMagic (fromInteger magicNbr))
```

Comment:
Must: map mainnet network magic to `Mainnet` like `fromNetworkMagic` from `cardano-api` does